src/main/java/org/benetech/servicenet/adapter/healthleads/persistence/PersistanceManager.java

```diff
@@ -0,0 +1,198 @@
+package org.benetech.servicenet.adapter.healthleads.persistence;
+
+import org.apache.commons.lang3.StringUtils;
+import org.benetech.servicenet.adapter.healthleads.HealthLeadsDataMapper;
+import org.benetech.servicenet.adapter.healthleads.model.HealthleadsBaseData;
+import org.benetech.servicenet.adapter.healthleads.model.HealthleadsEligibility;
+import org.benetech.servicenet.adapter.healthleads.model.HealthleadsLanguage;
+import org.benetech.servicenet.adapter.healthleads.model.HealthleadsLocation;
+import org.benetech.servicenet.adapter.healthleads.model.HealthleadsOrganization;
+import org.benetech.servicenet.adapter.healthleads.model.HealthleadsPhone;
+import org.benetech.servicenet.adapter.healthleads.model.HealthleadsPhysicalAddress;
+import org.benetech.servicenet.adapter.healthleads.model.HealthleadsRequiredDocument;
+import org.benetech.servicenet.adapter.healthleads.model.HealthleadsService;
+import org.benetech.servicenet.adapter.healthleads.model.HealthleadsServiceTaxonomy;
+import org.benetech.servicenet.adapter.healthleads.model.HealthleadsTaxonomy;
+import org.benetech.servicenet.adapter.shared.model.storage.EntryDictionary;
+import org.benetech.servicenet.domain.DataImportReport;
+import org.benetech.servicenet.domain.Eligibility;
+import org.benetech.servicenet.domain.Language;
+import org.benetech.servicenet.domain.Location;
+import org.benetech.servicenet.domain.Organization;
+import org.benetech.servicenet.domain.Phone;
+import org.benetech.servicenet.domain.PhysicalAddress;
+import org.benetech.servicenet.domain.RequiredDocument;
+import org.benetech.servicenet.domain.Service;
+import org.benetech.servicenet.domain.ServiceTaxonomy;
+import org.benetech.servicenet.manager.ImportManager;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class PersistanceManager {
+
+    private EntryDictionary<HealthleadsBaseData> dictionary = new EntryDictionary<>();
+    private HealthLeadsDataMapper mapper;
+    private ImportManager importManager;
+
+    private static final String PROVIDER_NAME = "healthleads";
+
+    public PersistanceManager(ImportManager importManager) {
+        this.mapper = HealthLeadsDataMapper.INSTANCE;
+        this.importManager = importManager;
+    }
+
+    public void addData(HealthleadsBaseData data) {
+        if (isLocationBased(data)) {
+            dictionary.addData(data.getLocationId(), data, HealthleadsLocation.class);
+        } else if (isServiceBased(data)) {
+            dictionary.addData(data.getServiceId(), data, HealthleadsService.class);
+        } else if (isOrganizationBased(data)) {
```

Comment:
are you sure of using 'else if' instead of 'if'? Do isLocationBased, isLocationBased and isOrganizationBased cases are always disjointed?

Reply:
We're not sure of that, but each data we've seen was constructed this way.